.github/workflows/lint_and_test.yml

```diff
@@ -0,0 +1,42 @@
+name: Lint and Test
+
+on: [push, pull_request]
+
+jobs:
+  fmt:
+    name: fmt
+    runs-on: ubuntu-latest
+    steps:
+      - uses: dtolnay/rust-toolchain@stable
```

Comment:
Cargo version is upgraded very quickly in Github images, not sure we need to bother with this

Having smart cargo cache with something like Swatinem/rust-cache might be more interesting